metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1417,6 +1543,7 @@ def emit_custom_sql_datasources(self) -> Iterable[MetadataWorkUnit]:
 
             datasource_name = None
             project = None
+            columns: List[Dict[Any, Any]] = []
```

Comment:
This is an issue I fixed along the way. We had ingestion runs that failed because the variable columns was not defined but used further down (in `schema_metadata = self.get_schema_metadata_for_custom_sql(columns)`). Before, it was only defined `if len(csql[c.DATA_SOURCES]) > 0` and for some reason that wasn't always the case for us. Let me know what you think.

Reply:
Looks reasonable - thanks for fixing that